tests/ui/lint/command-line-allow-warnings.rs

```diff
@@ -0,0 +1,27 @@
+// compile-flags: -A warnings
+
+fn main() {
+    // This will be overriden by the `-A warnings` command line option.
+    #[warn(non_snake_case)]
+    let _OwO = 0u8;
+
+    // But this should not.
+    #[deny(non_snake_case)]
+    let _UwU = 0u8;
+    //~^ ERROR variable `_UwU` should have a snake case name
+
+    bar();
+    baz();
+}
+
+#[warn(warnings)]
+fn bar() {
+    let _OwO = 0u8;
+    //~^ WARN variable `_OwO` should have a snake case name
```

Comment:
I don't understand how this work.
In this case in-source annotations take priority over command line option, but above (`#[warn(non_snake_case)]`) command line takes priority over in-source annotation.

Also, what `-A warnings` means exactly
- allow all lints (not true according to this test)
- allow lints with default level `warn`
- allow lints with effective level `warn` (explicit level with fallback to default)

?

Reply:
> In this case in-source annotations take priority over command line option, but above (`#[warn(non_snake_case)]`) command line takes priority over in-source annotation.

The `warnings` lint is... special. To the best of my knowedge, `-A warnings` suppresses warn level lints even when there is a source code annotation for the specific non-`warnings` lint (of warn level). However, when there is a `warnings` source code annotation, it overrides `-A warnings` for the lexical scope affected by the annotation.

Reply:
Also, lints with levels higher than `warn` are not affected by `-A warnings` or `#[allow(warnings)]` at all, such as deny-level lints.